alacritty/src/display/hint.rs

```diff
@@ -382,18 +401,19 @@ pub fn highlighted_at<T>(
         if let Some((hyperlink, bounds)) =
             hint.content.hyperlinks.then(|| hyperlink_at(term, point)).flatten()
         {
-            return Some(HintMatch {
-                bounds,
-                action: hint.action.clone(),
-                hyperlink: Some(hyperlink),
-            });
+            let id = term.grid_mut()[*bounds.end()].id();
+            let hint =
+                HintMatch { bounds, action: hint.action.clone(), hyperlink: Some(hyperlink) };
+            return Some(HighlightedHint::new(hint, id));
         }
 
         let bounds = hint.content.regex.as_ref().and_then(|regex| {
             regex.with_compiled(|regex| regex_match_at(term, point, regex, hint.post_processing))
         });
         if let Some(bounds) = bounds.flatten() {
-            return Some(HintMatch { bounds, action: hint.action.clone(), hyperlink: None });
+            let id = term.grid_mut()[*bounds.end()].id();
+            let hint = HintMatch { bounds, action: hint.action.clone(), hyperlink: None };
```

Comment:
I notice you're not using the `new` method here. Perhaps changing it to be `fn new(hint: HintMatch, term: &mut Term) -> Self` and moving the id extraction in there?

Reply:
Yeah I don't see why not tbh.